salt/output/json_out.py

```diff
@@ -52,6 +56,13 @@ def output(data, **kwargs):  # pylint: disable=unused-argument
     '''
     Print the output data in JSON
     '''
+    log_dict = _get_log_stream_as_dict()
+    # check if log entries exists.
+    #   try to avoid issues in ssh communication by salt.client.ssh handle_ssh()
+    #   could be checked by `tests/runtests.py -C --ssh`
+    if len(log_dict['logs']) > 0:
```

Comment:
I'm not a fan of this condition. I would be thankful for each other solution to avoid broken ssh communication in `tests/runtests.py -C --ssh` without this line.